airbyte-integrations/bases/base-singer/base_singer/source.py

```diff
@@ -68,3 +68,61 @@ def read(
 
         read_cmd = self.read_cmd(logger, config_container.rendered_config_path, selected_singer_catalog_path, state_path)
         return SingerHelper.read(logger, read_cmd)
+
+
+class BaseSingerSource(SingerSource):
+    force_full_refresh = False
+
+    def discover_cmd(self, logger: AirbyteLogger, config_path: str) -> str:
+        return f"{self.tap_cmd} --config {config_path} --discover"
+
+    def read_cmd(self, logger: AirbyteLogger, config_path: str, catalog_path: str, state_path: str = None) -> str:
+        state_path = None if self.force_full_refresh else state_path
+        args = {"--config": config_path, "--catalog": catalog_path, "--state": state_path}
+        cmd = " ".join([f"{k} {v}" for k, v in args.items() if v is not None])
+
+        return f"{self.tap_cmd} {cmd}"
+
+    def check(self, logger: AirbyteLogger, config_container: ConfigContainer) -> AirbyteConnectionStatus:
+        try:
+            json_config = config_container.rendered_config
+            self.try_connect(logger, json_config)
+        except self.api_error as err:
+            logger.error("Exception while connecting to the %s: %s", self.tap_name, str(err))
+            # this should be in UI
+            error_msg = (
+                f"Unable to connect to the {self.tap_name} with the provided credentials. "
```

Comment:
I can see name being e.g: "Postgres" which would make the phrasing awkward, just removing "the" to prevent this
```suggestion
                f"Unable to connect to {self.tap_name} with the provided credentials. "
```